CONTRIBUTING.md

```diff
@@ -128,6 +128,86 @@ Two other things to keep in mind:
 - If you add code that should be tested, add tests.
 - If you change the public API, [update the documentation](#api-reference).
 
+### Debugging Environment
+
+Due to the way that python and rust interoperate, debugging the Rust side of development from Python calls can be difficult. For this reason, we have provided
+a guide and several tools to setup up [Visual Studio Code](https://code.visualstudio.com/) with a debugging environment that makes debugging relatively simple.
+
+#### VS Code Extensions
+
+Install The [CodeLLDB](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) and the [Task Shell Input](https://marketplace.visualstudio.com/items?itemName=augustocdias.tasks-shell-input) extensions.
+You can install from within a VS Code terminal with the following (use `-r` to run in the current VSCode window):
+
+```shell
+code -r --install-extension vadimcn.vscode-lldb augustocdias.tasks-shell-input
+code -r --install-extension augustocdias.tasks-shell-input
+```
+
+#### Debug Configuration
+
+Copy the following `configurations` and `inputs` components to your `launch.json` file.
+
+<details><summary><code><b>launch.json</b></code></summary>
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Debug Rust",
+            "type": "python",
+            "request": "launch",
+            "program": "${workspaceFolder}/py-polars/debug/launch.py",
+            "args": [
+                "${file}",
+            ],
+            "console": "internalConsole",
+            "justMyCode": true,
+            "serverReadyAction": {
+                "pattern": "pID = ([0-9]+)",
+                "action": "startDebugging",
+                "name": "Rust LLDB",
+            },
+        },
+        {
+            "name": "Rust LLDB",
+            "type": "lldb",
+            "request": "attach",
+            "program": "${workspaceFolder}/py-polars/.venv/bin/pytho",
```

Comment:
Should there be a `n` at the end of python here?

Reply:
Yup, I thought I added that back. Will fix.